app/controllers/topic_resource_controller.rb

```diff
@@ -0,0 +1,74 @@
+require 'erb'
+require 'sinatra/flash'
+require 'pony'
+require 'json'
+
+module Sinatra
+  module App
+    module TopicResourceController
+
+      def self.registered(app)
+        app.get '/admin/topic/:id/resources' do
+          check_authentication
+          @resources = Resource.all
+          @topic = Topic.find_by(id: params[:id])
+          @topic_resources = @topic.resources
+          @title = 'Topic Resources'
+          erb :topic_resources
+        end
+
+        app.delete '/admin/topic/:id/resource/:resource_id/delete' do
+          check_admin_authentication
+          @topic = Topic.find(params[:id])
+          id = params[:id]
+          @topic_resources = @topic.resources
+          @resource = @topic.resources.find(params[:resource_id])
+          if @topic_resources.delete(@resource)
+            flash[:success] = 'Successfully remove the resource.'
+            redirect '/admin/topic/' + id + '/resources'
+          else
+            flash[:halt] = 'Unable to delete the topic'
+            redirect '/admin/topic/' + id + '/resources'
+          end
+        end
+
+        app.get '/admin/topic/:id/resource/:resource_id' do
+          check_admin_authentication
+          @topic = Topic.find(params[:id])
+          id = params[:id]
+          @topic_resources = @topic.resources
+          @resource = @topic.resources.find(params[:resource_id])
+          if @resource
+            erb :resource
+          else
+            flash[:halt] = 'Resource could not be found.'
+            redirect '/admin/topic/' + id + '/resources'
+          end
+        end
+
+        app.post '/admin/topic/:id/resources' do
```

Comment:
For this route handler, it is clear that you are adding a resource to a topic. However, most of the logic here can be pushed down to the model. As a rule, prefer to push business logic down to the model.

For example, you are validating that an existing topic resource is not added to the topic. Typically, validations are done in the model. 

Just a suggestion, you could have a method in your `Topic` model, say `add_resource`, that handles adding resources to a Topic.